### PR TITLE
Home Page Backend Pull Request

### PR DESCRIPTION
**Pull request by fullernle**

### CODE REVIEW

File: Server PHP/accountFunctions.php [hunk 1]
```diff
@@ -67,23 +92,49 @@ function loginAccount($username, $password, $sock) {
...
+    $username = $user->getName();
+    echo "DEBUG: Changing $username to offline \n";
+    clearOnlineStatus($username);
+    if (count($user->getCurrGroups()) != 0) {
+      echo "DEBUG: Removing $username from groups \n";
+      clearFromGroup($user);
+    }
+    $user->setName(NULL);
+    $user->setEmail(NULL);
+    $user->clearGroups();
```
Comment (ParkerD559) [inline Server PHP/accountFunctions.php]: Shouldn't clearOnlineStatus() do these things?

File: Server PHP/classes.php [hunk 2]
```diff
@@ -46,16 +51,117 @@ function setSocket($sock) {
...
+  }
+  function setRecGroup($groupID) {
+    if (in_array($groupID, $this->recGroups)) {
+      $key = array_search($groupID, $this->recGroups);
+      if ($key != 0){
+        $count = count($this->recGroups);
+        echo "DEBUG: Trying to add to Recent Group this is the count before add: $count\n";
+        if(count($this->recGroups) >= $this->numRecent){
+          array_unshift($this->recGroups, $groupID);
+          $this->recGroups = array_splice($this->recGroups, 0, $this->numRecent);
```
Comment (ParkerD559) [inline Server PHP/classes.php]: the 0 param isnt needed

File: Server PHP/classes.php [hunk 3]
```diff
@@ -46,16 +51,117 @@ function setSocket($sock) {
...
+        }
+        else
+          array_unshift($this->recGroups, $groupID);
+      }
+    }//if in array
+    else {
+      if(count($this->recGroups) >= $this->numRecent) {
+        array_unshift($this->recGroups, $groupID);
+        $this->recGroups = array_splice($this->recGroups, 0, 5);
```
Comment (ParkerD559) [inline Server PHP/classes.php]: should the 5 be numrecent?

File: Server PHP/classes.php [hunk 4]
```diff
@@ -46,16 +51,117 @@ function setSocket($sock) {
...
+    else
+      array_unshift($this->favGroups, $groupID);
+  }
+  function removeFavGroup($groupID) {
+    if (in_array($groupID, $this->favGroups)) {
+      $key = array_search($groupID, $this->favGroups);
+      array_splice($this->favGroups, $key, 1);
+    }
   }
   function removeGroup($groupID) {
```
Comment (ParkerD559) [inline Server PHP/classes.php]: Probably rename to removeRecentGroup?

File: Server PHP/classes.php [hunk 5]
```diff
@@ -46,16 +51,117 @@ function setSocket($sock) {
...
+    return $this->memberIPs;
+  }
+  function getAdmin() {
+    return $this->admin;
+  }
+  //Set Functions
+  function setGroupID($groupname) {
+      $this->groupID = $groupname;
+  }
+  function setMember($member) {
```
Comment (ParkerD559) [inline Server PHP/classes.php]: I think renaming to addMember would be clearer, and should this work with the user objects so it adds not just the name but the actual user?

File: Server PHP/groupFunctions.php [hunk 6]
```diff
@@ -103,24 +136,23 @@ function joinGroup($groupID, $user, $clientList, $sock)
...
             echo "DEBUG: No whiteboard in Database or placeholder, nothing to send...\n\n";
-            $bool_check = true;
           }
           else {
             $message = "WBUP$groupID $wbstring";
             echo "DEBUG: Whiteboard found in Database, updating for first user...\n\n";
             sendMessage($message, $sock);
-            $bool_check = true;
           } // closes line 89 else statement
         }   // closes line 82 numUsers if statement
```
Comment (ParkerD559) [inline Server PHP/groupFunctions.php]: all these comments arent even correct anymore -_-

**Comment by fullernle** [comment]
Updated code in response to suggestions, thoughts?

**Comment by fullernle** [comment]
Yeah, the name changes are more descriptive of what the function is actually affecting.  Any function that directly updates the database now has DB attached to the end of it, where as functions that only affect the global group list will have GL, and the global client list will have CL.

File: Server PHP/classes.php (hunk 2 above)
Comment (ParkerD559) [reply Server PHP/classes.php]: well nevermind apparently #justphpthings

**Comment by ParkerD559** [comment]
Is there gonna be more changes to this

**Pull request merged**